main.go

```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/jackc/pgx/v4/pgxpool"
+	repository "github.com/serhiihuberniuk/blog-api/repository/postgresql"
+	"github.com/serhiihuberniuk/blog-api/service"
+	"github.com/serhiihuberniuk/blog-api/view/handlers"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+)
+
+const dbUrl = "postgres://serhii:serhii@localhost:5432/api"
+
+func PostgresConnPool(ctx context.Context, dbUrl string) (*pgxpool.Pool, error) {
+	pool, err := pgxpool.Connect(ctx, dbUrl)
+	if err != nil {
+		return nil, fmt.Errorf("connection to database failed: %w", err)
+	}
+
+	return pool, nil
+}
+
+func main() {
+	ctx := context.Background()
+
+	pool, err := PostgresConnPool(ctx, dbUrl)
+	if err != nil {
+		log.Fatalf("cannot connect to DB: %v", err)
+	}
+
+	repo := &repository.Repository{
+		Db: pool,
+	}
+
+	serv := service.NewService(repo)
+
+	handler := handlers.NewHandlers(serv)
+
+	srv := http.Server{
+		Addr:    ":8080",
+		Handler: handler.ApiRouter(),
+	}
+
+	go func() {
+		if err := http.ListenAndServe(srv.Addr, srv.Handler); err != nil {
+			log.Fatalf("error while starting server: %v", err)
```

Comment:
the problem is that if the HTTP service won't start, you will fatal goroutine, so the main goroutine fails also and the connection pool will not be closed in a good way.